File: liveroom-core/src/main/java/com/qlive/core/QRooms.java
package com.qlive.core;

import com.qlive.core.been.QCreateRoomParam;
import com.qlive.core.been.QLiveRoomInfo;

import java.util.List;

/**
 * 房间管理接口
 */
public interface QRooms {

    /**
     * 创建房间
     *
     * @param param 创建房间参数
     * @param callBack
     */
    void createRoom(QCreateRoomParam param, QLiveCallBack<QLiveRoomInfo> callBack);

    /**
     * 删除房间
     * @param roomID  房间ID
     * @param callBack
     */
    void deleteRoom(String roomID, QLiveCallBack<Void> callBack);

    /**
     * 房间列表
     *
     * @param pageNumber
     * @param pageSize
     * @param callBack
     */
    void listRoom(
            int pageNumber,
            int pageSize,
            QLiveCallBack<List<QLiveRoomInfo>> callBack
    );

    /**
     * 根据ID获取房间信息
     *
     * @param roomID 房间ID
     * @param callBack
     */
    void getRoomInfo(String roomID, QLiveCallBack<QLiveRoomInfo> callBack);
}

File: liveroom-core/src/main/java/com/qlive/core/QRooms.kt
package com.qlive.core

import com.qlive.core.been.QCreateRoomParam
import com.qlive.core.been.QLiveRoomInfo

/**
 * 房间管理接口
 * @constructor Create empty Q rooms
 */
interface QRooms {

    /**
     * 创建房间
     *
     * @param param 创建房间参数
     * @param callBack
     */
    fun createRoom(param: QCreateRoomParam, callBack: QLiveCallBack<QLiveRoomInfo>?) //创建房间

    /**
     * 删除房间
     *
     * @param roomID
     * @param callBack
     */
    fun deleteRoom(roomID: String, callBack: QLiveCallBack<Void>?) //删除房间

    /**
     * 房间列表
     *
     * @param pageNumber
     * @param pageSize
     * @param callBack
     */
    fun listRoom(
        pageNumber: Int,
        pageSize: Int,
        callBack: QLiveCallBack<List<QLiveRoomInfo>>?
    ) //房间列表

    /**
     * 根据ID获取房间信息
     *
     * @param roomID 房间ID
     * @param callBack
     */
    fun getRoomInfo(roomID: String, callBack: QLiveCallBack<QLiveRoomInfo>?) //根据ID搜索房间信息

}